### PR TITLE
Feature infinity binning

### PR DESCRIPTION
**Pull request by dwlueth**
Fixes #
-  Infinity Numeric binning working against simulator
## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: open-sphere-base/mantle/src/main/java/io/opensphere/mantle/infinity/QueryParameters.java [hunk 1]
```diff
@@ -121,6 +130,47 @@ public void setBinFieldType(Class<?> binFieldType)
         myBinFieldType = binFieldType;
     }
 
+
+    /**
+     * Gets the bin's width
```
Comment (rkausch) [inline open-sphere-base/mantle/src/main/java/io/opensphere/mantle/infinity/QueryParameters.java]: End the Javadoc with a period (doccheck complains if it cannot find a complete sentance). Do for all javadocs here.

File: open-sphere-plugins/infinity/src/main/java/io/opensphere/infinity/envoy/InfinityEnvoy.java [hunk 2]
```diff
@@ -237,8 +237,18 @@ private SearchRequest createSearchRequest(QueryParameters parameters)
                     || DynamicEnumerationKey.class.isAssignableFrom(parameters.getBinFieldType()))
             {
                 field += ".keyword";
+                request.setAggs(new Aggs(field, 10000, InfinityUtilities.MISSING_VALUE));
```
Comment (rkausch) [inline open-sphere-plugins/infinity/src/main/java/io/opensphere/infinity/envoy/InfinityEnvoy.java]: Can you put the number into a constant? It's difficult to discern why 10,000 is the chosen value for the defaults.

Comment (rkausch) [reply open-sphere-plugins/infinity/src/main/java/io/opensphere/infinity/envoy/InfinityEnvoy.java]: (and it looks like it was that way before your changes)

File: open-sphere-plugins/infinity/src/main/java/io/opensphere/infinity/json/Aggs.java [hunk 3]
```diff
@@ -25,6 +25,19 @@ public Aggs(String field, int size, long missing)
         myBins = new Bins(field, size, missing);
     }
 
+    /**
+     * Constructor.
+     *
+     * @param field
```
Comment (rkausch) [inline open-sphere-plugins/infinity/src/main/java/io/opensphere/infinity/json/Aggs.java]: document parameters

File: open-sphere-plugins/infinity/src/main/java/io/opensphere/infinity/json/Bins.java [hunk 4]
```diff
@@ -48,6 +64,26 @@ public void setTerms(Terms terms)
         myTerms = terms;
     }
 
+    /**
+     * Get the histogram
```
Comment (rkausch) [inline open-sphere-plugins/infinity/src/main/java/io/opensphere/infinity/json/Bins.java]: punctuation

File: open-sphere-plugins/infinity/src/main/java/io/opensphere/infinity/json/Bucket.java [hunk 5]
```diff
@@ -1,10 +1,10 @@
 package io.opensphere.infinity.json;
 
 /** Elasticsearch bucket JSON bean. */
-public class Bucket
+public class Bucket<T>
```
Comment (rkausch) [inline open-sphere-plugins/infinity/src/main/java/io/opensphere/infinity/json/Bucket.java]: Document the type parameter in the class javadoc (e.g.: @param <T> something something

File: open-sphere-plugins/infinity/src/main/java/io/opensphere/infinity/json/Histogram.java [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    private long myMissing;
+
+    /** The bin offset */
+    private double myOffset;
+
+    /**
+     * Constructor
+     */
+    public Histogram()
+    {
```
Comment (rkausch) [inline open-sphere-plugins/infinity/src/main/java/io/opensphere/infinity/json/Histogram.java]: Document that this is intentionally blank (comment within the empty body)

File: open-sphere-plugins/infinity/src/test/java/io/opensphere/infinity/simulator/InfinitySimulator.java [hunk 7]
```diff
@@ -37,10 +50,29 @@ protected void handle(HttpExchange exchange) throws IOException
...
+                myBinOffset = request.getAggs().getBins().getHistogram().getOffset();
+                myMinDocCount = request.getAggs().getBins().getHistogram().getMinDocCount();
+                aggsField = request.getAggs().getBins().getHistogram().getField();
+                writeResponse(exchange, HttpURLConnection.HTTP_OK, getResponseBody(aggsField));
+            }
+        }
 
         exchange.getResponseHeaders().add("Content-Type", "application/json");
-        writeResponse(exchange, HttpURLConnection.HTTP_OK, getResponseBody(aggsField));
+
```
Comment (rkausch) [inline open-sphere-plugins/infinity/src/test/java/io/opensphere/infinity/simulator/InfinitySimulator.java]: Remove extra blank line

File: open-sphere-plugins/infinity/src/main/java/io/opensphere/infinity/json/Histogram.java (hunk 6 above)
Comment (dwlueth) [reply open-sphere-plugins/infinity/src/main/java/io/opensphere/infinity/json/Histogram.java]: Updated.  I am not sure why this isn't showing here. https://github.com/ngageoint/opensphere-desktop/commit/3dafa702cd2fe9527121d18c0b971ddd2654ba12#diff-6

**Pull request merged**